__tests__/app.test.js

```diff
@@ -185,3 +185,46 @@ describe('GET /api/articles/:article_id/comments', () => {
     });
   });
 });
+
+describe.only('POST /api/articles/:article_id/comments', () => {
+  test('Status 201: Request body accepts an object with the following properties: username, body and responds with: the posted comment', ()=>{
+    return request(app).post('/api/articles/3/comments').send({username: 'rogersop', body: 'your article looks really nice'}).expect(201).then(({body})=>{
+      const commentObj = {
+        comment_id: expect.any(Number),
+        body: 'your article looks really nice',
+        article_id: 3,
+        author: 'rogersop',
+        votes: expect.any(Number),
+        created_at: expect.any(String),
+      };
+      expect(body.comment).toEqual(expect.objectContaining(commentObj));
+    });
+  });
+  test('Status 404: should respond with ""article_id" is an invalid Article ID." when an incorrect article id is passed', ()=>{
+    return request(app)
+        .post('/api/articles/10101/comments')
+        .send({username: 'rogersop', body: `this article doesn't even exist`})
+        .expect(404)
+        .then(({body})=>{
+          expect(body.msg).toBe(`10101 is an invalid Article ID.`);
+        });
+  });
+  test('Status 400: should respond with "Invalid Input" when given an invalid format within the request body', ()=>{
+    return request(app)
+        .post('/api/articles/1/comments')
+        .send({username: 'rogersop', handbag: 123})
+        .expect(400)
+        .then(({body})=>{
+          expect(body.msg).toBe(`Invalid Input`);
+        });
+  });
+  test('Status 400: responds with "Username Does Not Exist" when the request includes a username not in the database', ()=>{
```

Comment:
what status is used when something can't be found?

Reply:
404! 🤦